auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,376 @@
+                        "\nYour IP address seems to be too short.\n")
+                messageContent += _("Here is what we got instead:\n")
+        if not conn_success:
+            raise except_orm(
+                _(messageTitle), _(
+                    messageContent + "%s") %
+                tools.ustr(e))
+        else:
+            raise UserError(_(messageTitle), _(messageContent))
+
+    @api.model
+    def schedule_backup(self):
+
+        for rec in self.search([]):
+            if not os.path.isdir(rec.bkp_dir):
+                os.makedirs(rec.bkp_dir)
+            # Create name for dumpfile.
+            bkp_file = '%s_%s.dump.zip' % (
+                time.strftime('%d_%m_%Y_%H_%M_%S'),
+                rec.name)
+            file_path = os.path.join(rec.bkp_dir, bkp_file)
+            fbk = open(file_path, 'wb')
+            db.dump_db(rec.name, fbk)
+            fbk.close()
+            # Check if user wants to write to SFTP or not.
+            if rec.sftpwrite is True:
+                try:
+                    # Store all values in variables
+                    dir = rec.bkp_dir
+                    pathToWriteTo = rec.sftppath
+                    ipHost = rec.sftpip
+                    portHost = rec.sftpport
+                    usernameLogin = rec.sftpusername
+                    passwordLogin = rec.sftppassword
+                    # Connect with external server over SFTP
+                    srv = pysftp.Connection(
+                        host=ipHost,
+                        username=usernameLogin,
+                        password=passwordLogin,
+                        port=portHost)
+                    # Move to the correct directory on external server. If the
+                    # user made a typo in his path with multiple slashes
+                    # (/odoo//backups/) it will be fixed by this regex.
+                    pathToWriteTo = re.sub('/+', '/', pathToWriteTo)
+                    _logger.debug(
+                        'Start to copy files..'
+                    )
+                    try:
+                        srv.chdir(pathToWriteTo)
+                    except IOError:
+                        # Create directory and subdirs if they do not exist.
+                        currentDir = ''
+                        for dirElement in pathToWriteTo.split('/'):
+                            currentDir += dirElement + '/'
+                            try:
+                                srv.chdir(currentDir)
+                            except:
+                                _logger.info(
+                                    _(
+                                        '(Part of the) path didn\'t exist. '
+                                        'Creating it now at %s'
+                                    ) % currentDir
+                                )
+                                # Make directory and then navigate into it
+                                srv.mkdir(currentDir, mode=777)
```

Comment:
Why would a sysadmin want that my database dumps are readable by everyone by default? I already own the folder, so use a stricter mode please.